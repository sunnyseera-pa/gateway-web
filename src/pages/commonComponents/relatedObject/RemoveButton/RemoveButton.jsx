/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import SVGIcon from '../../../../images/SVGIcon';
import * as styles from './RemoveButton.styles';
import '../../CommonComponents.scss';
import '../RelatedObject.scss';

const RemoveButton = ({ removeButtonHandler }) => {
	return (
		<Button variant='medium' className='soft-black-14' onClick={removeButtonHandler} data-testid='remove-button'>
			<SVGIcon name='closeicon' fill={'#979797'} className='mr-2' css={styles.buttonSvg} />
			Remove
		</Button>
	);
};

RemoveButton.propTypes = {
	removeButtonHandler: PropTypes.func.isRequired,
};

export default RemoveButton;
