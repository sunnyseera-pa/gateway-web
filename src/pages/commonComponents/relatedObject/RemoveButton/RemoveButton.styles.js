import { css } from '@emotion/react';

export const buttonSvg = css`
	width: 10px;
	height: 10px;
`;
