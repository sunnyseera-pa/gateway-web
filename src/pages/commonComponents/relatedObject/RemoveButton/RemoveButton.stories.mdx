import { Meta, Story, Canvas } from '@storybook/addon-docs';
import RemoveButton from './RemoveButton';

<Meta title='Related Resources' component={RemoveButton} />

<Canvas>
	<Story name='Remove Button'>
		{props => {
			const removeButton = () => {
				alert('removed');
			};
			return <RemoveButton {...props} removeButtonHandler={removeButton} />;
		}}
	</Story>
</Canvas>
