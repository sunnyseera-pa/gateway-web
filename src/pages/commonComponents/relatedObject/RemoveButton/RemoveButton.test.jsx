import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import RemoveButton from './RemoveButton';

const props = {
    removeButtonHandler: jest.fn(),
};
let wrapper;

describe('Given the Remove Button component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<RemoveButton {...props} />);
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then Button should be rendered with SVG icon', () => {
            expect(screen.getByTestId('remove-button')).toBeTruthy();
            expect(screen.getByTestId('closeicon')).toBeTruthy();
        });

        it('Then onclick should call removeButtonHandler function', () => {
            fireEvent.click(screen.getByTestId('remove-button'));
            expect(props.removeButtonHandler.mock.calls.length).toBe(1);
        });
    });
});
