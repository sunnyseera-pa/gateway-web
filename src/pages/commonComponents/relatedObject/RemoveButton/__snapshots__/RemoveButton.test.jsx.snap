// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Remove Button component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <button
    class="soft-black-14 btn btn-medium"
    data-testid="remove-button"
    type="button"
  >
    <svg
      class="mr-2 css-1cyke48-RemoveButton"
      data-testid="closeicon"
      height="100%"
      viewBox="0 0 18 18"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <path
        d="M19 3.58578644 20.4142136 5 13.415 12 20.4142136 19 19 20.4142136 12 13.415 5 20.4142136 3.58578644 19 10.585 12 3.58578644 5 5 3.58578644 12 10.585z"
        fill="#979797"
      />
    </svg>
    Remove
  </button>
</div>
`;
